PyComPWA.cpp

```diff
@@ -173,7 +173,48 @@ PYBIND11_MODULE(ui, m) {
       .def(py::init<std::vector<ComPWA::pid>, std::vector<ComPWA::Event>>(),
            py::arg("pids"), py::arg("events"))
       .def_readonly("pids", &ComPWA::EventCollection::Pids)
-      .def_readonly("events", &ComPWA::EventCollection::Events);
+      .def_readonly("events", &ComPWA::EventCollection::Events)
+      .def("to_table",
+           [](const ComPWA::EventCollection &self) {
+             if (!self.checkPidMatchesEvents())
+               throw ComPWA::CorruptFile(
+                   "Number of PIDs in EventCollection does not match the "
+                   "number of four momenta in each of its events");
+             std::vector<std::vector<double>> table;
+             for (const auto &Event : self.Events) {
+               std::vector<double> row;
+               row.reserve(self.Pids.size() * 4);
+               for (const auto &Momentum : Event.FourMomenta) {
+                 row.push_back(Momentum.px());
+                 row.push_back(Momentum.py());
+                 row.push_back(Momentum.pz());
+                 row.push_back(Momentum.e());
+               }
+               table.push_back(std::move(row));
+             }
+             return table;
+           })
+      .def("weights",
+           [](const ComPWA::EventCollection &self) {
+             std::vector<double> Weights;
+             Weights.reserve(self.Events.size());
+             for (const auto &Event : self.Events) {
+               Weights.push_back(Event.Weight);
+             }
+             return Weights;
+           })
+      .def("has_weights", [](const ComPWA::EventCollection &self) {
+        if (!self.Events.size()) {
+          return false;
+        }
+        auto FirstWeight = self.Events.front().Weight;
+        for (const auto &Event : self.Events) {
+          if (Event.Weight != FirstWeight) {
+            return true;
+          }
+        }
+        return false;
```

Comment:
Why do you need this as separate functions, I would add this functionality to `to_table`.

Reply:
See https://github.com/ComPWA/pycompwa/pull/78#discussion_r409806037

Reply:
So do I understand this correctly, that you can't but this "single" column into a multi index table?

Reply:
Can be done, but the column key would have to be e.g. `('weight', '')`. But this function was added so that a weight column is only added to the `DataFrame` if the `EventCollection` actually has weights (i.e. not all equal to `1.`)

Reply:
Ok its basically related to the problem with the EventCollection structure and Event as mentioned [here](https://github.com/ComPWA/pycompwa/pull/78/files#r410218578). I'm gonna resolve this, as pycompwa is going to be archived anyway